app/controllers/books_controller.rb

```diff
@@ -0,0 +1,12 @@
+class BooksController < ApplicationController
+  include Wor::Paginate
+  include DeviseTokenAuth::Concerns::SetUserByToken
+  def index
+    render_paginated Book.all
+  end
+
+  def shows
+    render json: Book.find(params[:id])
+  end
+  before_action :authenticate_user!
```

Comment:
As a convention, all `before_action` should be on top of the file, before any `def`.